src/lib.rs

```diff
@@ -105,13 +106,35 @@ impl Deref for RedactedString {
     }
 }
 
+/// Authentication method for Deepgram API requests.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum AuthMethod {
+    /// Use an API key with "Token" prefix (e.g., "Token dg_xxx").
+    /// This is for permanent API keys created in the Deepgram console.
+    ApiKey(RedactedString),
+
+    /// Use a temporary token with "Bearer" prefix (e.g., "Bearer dg_xxx").
+    /// This is for temporary tokens obtained via token-based authentication.
+    TempToken(RedactedString),
+}
+
+impl AuthMethod {
+    /// Get the authorization header value for this authentication method.
+    pub(crate) fn header_value(&self) -> Option<String> {
+        match self {
+            AuthMethod::ApiKey(key) => Some(format!("Token {}", key.0)),
+            AuthMethod::TempToken(token) => Some(format!("Bearer {}", token.0)),
+        }
+    }
```

Comment:
This always returns `Some()` so it can return a `String instead of an `Option<String>`.